components/properties/src/script.rs

```diff
@@ -176,13 +309,56 @@ impl<'data> ScriptExtensions<'data> {
     ///
     /// If `code_point` has Script_Extensions, then return the Script codes in
     /// the Script_Extensions. In this case, the Script property value
-    /// (normally Common or Inherited) is not included in the set.
+    /// (normally Common or Inherited) is not included in the [`ZeroSlice`].
     ///
     /// If c does not have Script_Extensions, then the one Script code is put
-    /// into the set and also returned.
+    /// into the [`ZeroSlice`] and also returned.
+    ///
+    /// If c is not a valid code point, then return an empty [`ZeroSlice`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_provider::prelude::*;
+    /// use icu_properties::provider::key;
+    /// use icu::properties::provider::ScriptExtensionsPropertyV1Marker;
+    /// use icu_properties::Script;
+    /// use icu::properties::script::ScriptExtensions;
+    /// use zerovec::ZeroVec;
+    ///
+    /// let provider = icu_testdata::get_provider();
     ///
-    /// If c is not a valid code point, then the one UNKNOWN code is put into
-    /// the set and also returned.
+    /// let payload: DataPayload<ScriptExtensionsPropertyV1Marker> = provider
+    ///     .load_payload(
+    ///         key::SCRIPT_EXTENSIONS_V1,
+    ///         &DataRequest {
+    ///             options: ResourceOptions::default(),
+    ///             metadata: Default::default(),
+    ///         },
+    ///     )
+    ///     .expect("The data should be valid")
+    ///     .take_payload()
+    ///     .expect("Loading was successful");
+    ///
+    /// let scx: &ScriptExtensions = &payload.get().data;
+    ///
+    /// assert_eq!(
+    ///     scx.get_script_extensions_val('𐓐' as u32).as_zerovec(), // U+104D0 OSAGE CAPITAL LETTER KHA
+    ///     ZeroVec::<Script>::alloc_from_slice(&[Script::Osage])
```

Comment:
Suggestion: compare via `vec` for the purposes of API docs

```suggestion
    ///     scx.get_script_extensions_val('𐓐' as u32).iter().collect::<Vec<Script>>(), // U+104D0 OSAGE CAPITAL LETTER KHA
    ///     vec![Script::Osage]
```